check_msid_status.py

```diff
@@ -13,102 +13,82 @@
-            status = check_status_neumeric(msid, val, ldict)
-            return status
-        except:
-            return 'GREEN'
+    except:
+        pass
 #
-#--- the value is letters
+#--- General Violation
 #
-    except:
-        if val in ['NaN', 'nan', '']:
-            return 'CAUTION'
-
-        elif msid in ['AOCONLAW', 'AOCPESTL', '4OBAVTMF', '4OBTOORF', 'COSCS128S',\
-                    'COSCS129S','COSCS130S', 'COSCS131S','COSCS132S','COSCS133S',\
-                    'COSCS107S','CORADMEN', 'CCSDSTMF', 'ACAFCT', 'AOFSTAR',\
-                    '2SHLDART', 'PLINE03T', 'PLINE04T', 'AACCCDPT', '3LDRTNO']:
-            try:
-                status = check_status_letter(msid, val, vdict)
-                return status
-            except KeyError:
-                return 'GREEN'
-            except:
-                traceback.print_exc()
-                return 'GREEN'
-        else:
-            if val in ['ERR', 'FALT', 'FAIL']:
-                return "CAUTION"
-            else:
-                return 'GREEN'
+    if val in ['ERR', 'FALT', 'FAIL', 'NaN', 'nan', '']:
+        return 'CAUTION'
+#
+#--- Edge Case Violation
+#
+    elif msid in ['AOCONLAW', 'AOCPESTL', '4OBAVTMF', '4OBTOORF', 'COSCS128S',\
+                  'COSCS129S','COSCS130S', 'COSCS131S','COSCS132S','COSCS133S',\
+                  'COSCS107S','CORADMEN', 'CCSDSTMF', 'ACAFCT', 'AOFSTAR', '2SHLDART']:
+        return check_status_edge_case(msid, val, vdict)
+#
+#--- Numerical Violation
+#
+    elif isinstance(val, float):
+        return check_status_numeric(msid, val, ldict)
+#
+#--- Unchecked Instance
+#
+    else:
+        return 'GREEN'
 
 #-------------------------------------------------------------------------------
-#-- check_status_neumeric: check status of msid with nemeric value            --
+#-- check_status_numeric: check status of msid with numeric value            --
 #-------------------------------------------------------------------------------
 
-def check_status_neumeric(msid, val, ldict):
+def check_status_numeric(msid, val, ldict):
     """
-    check status of msid with nemeric value
+    check status of msid with numeric value
     input:  msid    --- msid
             val     --- current value of msid
-            ldict   --- dictionary of limits for nemeric entry
+            ldict   --- dictionary of limits for numeric entry
     output: status  --- current status of msid
     """
+    if msid in ldict.keys():
 #
-#--- if no condition, return 'GREEN'
+#--- condition exists; check agaist the value
 #
-    try:
         limit   = ldict[msid]
-    except:
-        return "GREEN"
-    
-    if (len(limit) == 0) or (limit == ['']):
-        return "GREEN"
+        if (val >= limit['warning_low']) and (val <= limit['warning_high']):
+            return "GREEN"
+        elif (val >= limit['caution_low']) and (val <= limit['caution_high']):
+            return "CAUTION"
+        else:
+            return "WARNING"
+    else:
```

Comment:
We store the limits as numeric values so converting them with float() is not necessary.

Reply:
OK